ext/ffi/jit_trampoline.rs

```diff
@@ -83,20 +96,35 @@ pub(crate) fn codegen(sym: &crate::Symbol) -> String {
     c += native_arg_to_c(ty);
     let _ = write!(c, " p{i}");
   }
+  if needs_unwrap {
+    let _ = write!(c, ", struct FastApiTypedArray* p_ret");
```

Comment:
I may be mistaken, but seems to me like it might be more appropriate to declare this as `struct FastApiTypedArray* const p_ret`, ie. that the pointer cannot be rewritten.